datatypes/src/spatial_reference.rs

```diff
@@ -119,6 +119,30 @@ impl SpatialReference {
     pub fn srs_string(&self) -> String {
         format!("{}:{}", self.authority, self.code)
     }
+
+    /// Compute the bounding box of this spatial reference that is also valid in the `other` spatial reference
+    #[allow(clippy::trivially_copy_pass_by_ref)]
+    pub fn valid_bounds<T>(&self, other: &SpatialReference) -> Result<T>
```

Comment:
i think it is fine to return a WGS84 bbox since the area of use is also wgs84

Reply:
do you mean returning the intersection of the spatial references in wgs84? But we need it in the projection of `self` in the ReprojectionOperator. We could do it as an extra step but then I guess this method here doesn't serve much purpose.